doc/guides/dts/index.rst

```diff
@@ -1,177 +1,249 @@
-  build is described in Kconfig. Through the use of Kconfig, users can opt not
-  to include support for this particular hardware item if they don't require it.
-
-Another example is a device with a 2.4GHz, multi-protocol radio supporting
-both the Bluetooth Low Energy and 802.15.4 wireless technologies. In this case:
-
-* Device Tree describes the presence of a radio peripheral compatible with a
-  certain radio driver.
-* Additional hardware boot-time configuration settings may also be present
-  in the Device Tree files. In this particular case it could be a
-  default TX power in dBm if the hardware does have a simple, cross-wireless
-  technology register for that.
-* Kconfig will describe which **protocol stack** is to be used with that radio.
-  The user may decide to select BLE or 802.15.4, which will both depend on
-  the presence of a compatible radio in the Device Tree files.
-
-Device tree file formats
-************************
+The first figure in the :ref:`device-tree-intro` shows how devicetree fits into
+the Zephyr build system. This section describes the input and output files in
+more detail.
+
+.. figure:: zephyr_dt_inputs_outputs.png
+   :figclass: align-center
+
+   Devicetree input (green) and output (yellow) files
+
+Devicetree source code files are written in the DTS format, which is defined in
+the `Devicetree specification`_. They usually have ``.dts`` or ``.dtsi`` (for
+Devicetree Source Include) extensions. Zephyr's build system looks for a file
+named :file:`BOARD.dts` in the board definition directory; this file contains
+the board's base devicetree. See :ref:`dt_k6x_example` for real-world examples.
+
+The build system combines the board's DTS with additional input files called
+*overlays* to produce a final devicetree source file. Overlays are also written
+in the DTS format, but have a :file:`.overlay` extension to make it clear that
+they're overlays. You can specify the overlay files to use at build time using
+the :makevar:`DTC_OVERLAY_FILE` CMake variable described in
+:ref:`important-build-vars`. The build system also looks for device tree
+overlays in several locations by default; see :ref:`application_dt` for the
+list.
+
+Overlays can be used to add or delete nodes from the tree, or to modify node
+properties and their values. Along with Kconfig, devicetree overlays let you
+reconfigure the kernel and device drivers without modifying their source code.
+
+Before they are combined, the C preprocessor is run on :file:`BOARD.dts` and any
+overlays. This allows these files to use C macros and include directives.
+
+The *devicetree compiler* tool, ``dtc``, is used to combine :file:`BOARD.dts`
+with any overlays into a final :file:`BOARD.dts_compiled` file in the
+application build directory.  :file:`BOARD.dts_compiled` is the final
+devicetree source. It is also in DTS format. More information on ``dtc`` can be
+found in the `dtc source code repository`_.
```

Comment:
> Maybe this could mention that the files are just concatenated (it's done here), and that multiple definitions for a node get merged in DTS.

This was also intentionally left out to keep PR scope smaller. This falls into the "more formally specify the bindings+dts --> scripts --> C process" follow-up work mentioned in the cover letter.

> The new script runs directly on the .dts files btw, and don't need dtc. The old scripts are run on the output of dtc -Odts <.dts file>, because they don't deal with things like replacing phandle (... = <&foo>;) and path (... = &foo;) references.

OK, I'll update the diagram, thanks.

> There's also a native DTS syntax for including files btw: /include/ foo. It doesn't seem to get used that often though.

Yep, I know. I didn't mention it precisely because it doesn't get used by Zephyr.